### DISCUSSION TITLE
set log level via configmap

### DISCUSSION THREAD

**Discussion by bulldog98** [quarkusio/quarkus#34364] [Q&A] [answered]
I use kubernetes-config to control some settings.

How do I configure some logger to use a level?
In `application.properties`
```
quarkus.log.category."com.ascm.SomeLogger".level=ERROR
```
If I want to put that into the ConfigMap I get
```
configmaps "some-cm" was not valid:
* data[quarkus.log.category."com.ascm.SomeLogger".level]: Invalid value: "quarkus.log.category.\"com.ascm.SomeLogger\".level": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')
```

Is there a way around that. Also is the logging configuration applied directly, in my tests it seemed to need a restart (even with application.properties)

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
If you are configuring something with `.properties`, you need to use something like `quarkus.log.category."com.ascm.SomeLogger".level= ERROR`

**Reply by bulldog98** [reply]
Yeah right, I copied the part I had in data in the config map over, sorry about that

**Reply by bulldog98** [reply]
The question was about howto translate that into a configmap mainly

**Reply by geoand** [reply]
A ConfigMap does support "files", so using that would be simple

**Reply by bulldog98** [reply, marked as answer]
Ah so I would have to do something like this?
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: some-cm
data:
  application.properties: |
     quarkus.log.category."com.ascm.SomeLogger".level=ERROR
```

and in the applications.properties in the container, I'd have:
```
# somewere namespace is also set
quarkus.kubernetes-config.enabled=true
quarkus.kubernetes-config.secrets.enabled=true
quarkus.kubernetes-config.config-maps=some-cm
quarkus.kubernetes-config.secrets=some-cm
```

Since atm I have the properties I want to set flat in the configmap

**Reply by geoand** [reply]
Yeah, that should work
